### PR TITLE
vestResolver

### PR DESCRIPTION
**Pull request by bluebill1049**
new resolver for Vest

https://github.com/ealush/vest

 - [ ] update doc

### CODE REVIEW

File: src/vest.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+              }, {}),
+            }
+          : {}),
+      },
+    };
+  }, {});
+};
+
+export const vestResolver = <TFieldValues extends FieldValues>(
+  schema: any,
```
Comment (jorisre) [inline src/vest.ts:31]: A schema _(validationSuite)_ is typed as `ICreateResult` but it not exposed, you can retrieve it by doing the following thing: 

```ts
type ICreateResult = ReturnType<typeof vest.create>;
```

**Review by jorisre — COMMENTED** [review]
👏🏻 

To expose `vestResolver` as the following `@hookform/resolvers/vest`.
You've to create two files:  `vest.js` &  `vest.d.ts` 

```js
// vest.js
module.exports = require('./dist/vest');
```

```ts
// vest.d.ts
export * from './dist/vest';
```
Examples [zod.js](https://github.com/react-hook-form/resolvers/blob/master/zod.js) & [zod.d.ts](https://github.com/react-hook-form/resolvers/blob/master/zod.d.ts)

Welcome to Vest in the @hookform's resolvers family :)

File: README.md [hunk 2]
```diff
@@ -157,7 +154,59 @@ const App = () => {
...
+    </form>
+  );
+};
+```
 
+### [Vest](https://github.com/ealush/vest)
+
+The most powerful data validation library for JS.
+
+[![npm](https://img.shields.io/bundlephobia/minzip/vest?style=for-the-badge)](https://bundlephobia.com/result?p=joi)
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply README.md:167]: oops

File: src/vest.ts (hunk 1 above)
Comment (bluebill1049) [reply src/vest.ts:31]: great suggestion

File: README.md [hunk 3]
```diff
@@ -157,7 +154,59 @@ const App = () => {
...
+  });
+
+  test('password', 'Password must contain a symbol', () => {
+    enforce(data.password).matches(/[^A-Za-z0-9]/);
+  });
+});
+
+const App = () => {
+  const { register, handleSubmit, errors } = useForm({
+    resolver: vestResolver,
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply README.md:203]: Thank you! 🙏

**Comment by bluebill1049** [comment]
Thanks very much for the review @jorisre 🙏

**Comment by jorisre** [comment]
@bluebill1049 To get it work, you must create 2 files under the root: `vest.js` & vest.d.ts`

And add them in package.json, under `files` property (like zod for example)

**Comment by bluebill1049** [comment]
> @bluebill1049 To get it work, you must create 2 files under the root: `vest.js` & vest.d.ts`
> 
> And add them in package.json, under `files` property (like zod for example)

oh oops sorry, I miss that comment. 🙏

**Review by jorisre — APPROVED** [review]
LGTM ! 👏🏻

**Review by ealush — COMMENTED** [review]
Very nice! So exciting. There are few things I noticed, but in general, this looks amazing!

File: package.json [hunk 4]
```diff
@@ -80,6 +80,7 @@
     "superstruct": "^0.8.3",
     "ts-jest": "^26.1.0",
     "typescript": "^3.7.5",
+    "vest": "^2.2.3",
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply package.json:85]: awesome!

File: src/vest.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    };
+  }, {});
+};
+
+export const vestResolver = <TFieldValues extends FieldValues>(
+  schema: ICreateResult,
+  _: any = {},
+  validateAllFieldCriteria = false,
+): Resolver<TFieldValues> => async (values) => {
+  const result = schema(values);
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply src/vest.ts:38]: awesome, let's go with yours!

File: README.md [hunk 6]
```diff
@@ -157,7 +154,59 @@ const App = () => {
...
+      <input type="submit" />
+    </form>
+  );
+};
+```
 
+### [Vest](https://github.com/ealush/vest)
+
+The most powerful data validation library for JS.
+
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply README.md:166]: I am keeping those notes on our internal V7 Trello board. :)

File: src/vest.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+  if (!Object.keys(errors).length) {
+    return { values: values as any, errors: {} };
+  }
+
+  return {
+    values: {},
+    errors: transformToNestObject(
+      parseErrorSchema(errors, validateAllFieldCriteria),
+    ),
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply src/vest.ts:49]: we probably going to introduce warning as well in the next major (it's in our board) 👍

File: src/vest.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+                  ...prev,
+                  [index]: message,
+                };
+              }, {}),
+            }
+          : {}),
+      },
+    };
+  }, {});
+};
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply src/vest.ts:38]: Shall we keep what we got for now? I think this is going to be a problem with nested values.

**Comment by bluebill1049** [comment]
@ealush Thanks for the review! I think we are ready to ship this next week if you are happy with it.

**Comment by bluebill1049** [comment]
looks like ealush is potentially on holiday, I will merge the PR by tomorrow and release it. 🚀  FYI @jorisre

**Pull request merged**